### DISCUSSION TITLE
How to create a reverse proxy?

### DISCUSSION THREAD

**Discussion by Macrisu** [NginxProxyManager/nginx-proxy-manager#2665] [Q&A] [answered]
Until now I used a Synology NAS and was easy to set up a reverse proxy for my docker apps.

My ISP is giving me a dynamic DNS, so my wan IP can be always accessed via xxx.go.ro

Usually, I had like this:

HTTPs://xxx.go.ro:8080 -> NAS IP: 8081 -> reverse proxy -> docker container:8080

I've installed Ngix Prix Manager on rpi + created an SSL certificate for xxx.go.ro

I've created a port fwd on my router
https://xxx.go.ro:8080 -> rPI:8081

How can I tell Nginx Proxy Manager, to **redirect any requests from 8081** to the docker container, port 8080?

**All the services are running on rpi Docker containers, and this seems to be the problem**

![how](https://user-images.githubusercontent.com/16325429/224007299-ad34563a-90a4-46a9-b98b-439af40b66fe.jpg)

**Comment by bmmmm** [comment, marked as answer]
If I got you right... __Nginx Proxy Manager__ and the other services are running on same machine (rpi)?
if yes, you can use the docker container names to forward your traffic.
e.g. add _nextcloud-app-1_ as **Forward Hostname / IP**

> How can I tell Nginx Proxy Manager, to redirect any requests from 8081 
you need to configure _Nginx Proxy Manager_ to listen on that port for HTTP? requests.

The domain field is needed to distinguish the service you want to address.

But... it might become very scatchi with all your port mappings... 

The idea of _Nginx Proxy Manager_ is  to send all your traffic to _Nginx Proxy Manager_ on default ports (80,443) and based on the domain, dispatch it to the corresponding service. 

**/edit/**
I just figgured out that it is already written here 😉: [https://nginxproxymanager.com/guide/#hosting-your-home-network](https://nginxproxymanager.com/guide/#hosting-your-home-network)

I suggest the following:
- if you are not able to create subdomains, get an own domain for 5 bucks a year or use a dyn dns provider to publish your IP (from your router/rpi) to the internet
- only forward port 80 and 443 from your router to rpi (_Nginx Proxy Manager_).
- _Nginx Proxy Manager_ listens to port 80 and 443.
- _Nginx Proxy Manager_ forwards based on domain / subdomains to the corresponding service to any port you want 😉.
- e.g. nextxloud.YOURDOMAIN.com , service2.YOURDOMAIN.com , service3.YOOURDOMAIN.com ...

**Reply by Macrisu** [reply]
Thank you! 
I have a lot of unused domains, so I've created a namecheap account for dynamic DNS.
I use Unifi UDM as a router and Namecheap is supported.

Tutorials for this:
https://www.namecheap.com/support/knowledgebase/article.aspx/583/11/how-do-i-configure-ddclient/
https://daltonflanagan.com/configuring-ubiquiti-unifi-usg-for-namecheap-ddns/
